sphinx/environment/__init__.py

```diff
@@ -296,6 +297,27 @@ def _config_status(*, old_config: Config | None, new_config: Config) -> tuple[in
                 extension = f'{len(extensions)}'
             return CONFIG_EXTENSIONS_CHANGED, f' ({extension!r})'
 
+        # Log any changes in configuration keys
+        if changed_keys := _differing_config_keys(old_config, new_config):
+            changed_num = len(changed_keys)
+            if changed_num == 1:
+                logger.info(
+                    __('1 configuration option has changed: %s'),
+                    changed_keys[0],
+                )
+            elif changed_num <= 10:
+                logger.info(
+                    __('%d configuration options have changed: %s'),
+                    changed_num,
+                    ', '.join(changed_keys),
```

Comment:
How about quoting each key? namely ', '.join(map(repr, changed_keys)) instead.

Reply:
Why? In conf.py, they're not quoted.

Reply:
Because it's a message and people could see the names better. For instance when reporting a bad parameter I usually quote that parameter name. But if you want to keep them unquoted I am also ok with it!